sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -171,13 +171,17 @@ case class Join(
 
   def selfJoinResolved: Boolean = left.outputSet.intersect(right.outputSet).isEmpty
 
-  // Joins are only resolved if they don't introduce ambiguous expression ids.
-  override lazy val resolved: Boolean = {
+  lazy val partlyResolved: Boolean = {
     childrenResolved &&
       expressions.forall(_.resolved) &&
       selfJoinResolved &&
       condition.forall(_.dataType == BooleanType)
   }
+  // Joins are only resolved if they don't introduce ambiguous expression ids.
```

Comment:
1. add a blank line above
2. we need to explain what the different resolution modes are.
3. I don't think partlyResolved is a good name -- it doesn't really explain anything, e.g. which part?